### PR TITLE
[New Rule] Persistence via Login and/or Logout Hooks

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/682

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+description = """
+Identifies use of the Defaults command to install a LoginHook Persistence. MacOS LoginHooks allow a script to be
+automatically executed at login and/or logout.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Persistence via Login and/or Logout Hooks"
+references = ["https://www.virusbulletin.com/uploads/pdf/conference_slides/2014/Wardle-VB2014.pdf"]
```
Comment (bm11100) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:17]: add man page https://www.manpagez.com/man/1/defaults/

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+risk_score = 47
+rule_id = "5d0265bf-dea9-41a9-92ad-48a8dcd05080"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ process.name : "defaults" and process.args : "write" and process.args : "LoginHook"
```
Comment (bm11100) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:26]: currently only looking for a login hook. Do you want to add `LogoutHook`? If so, can adjust the description / name appropriately.

Comment (Samirbous) [reply rules/macos/persistence_login_logout_hooks_defaults.toml:26]: good catch

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+[metadata]
+creation_date = "2020/12/07"
+maturity = "production"
+updated_date = "2020/12/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies use of the Defaults command to install a LoginHook Persistence. MacOS LoginHooks allow a script to be
+automatically executed at login and/or logout.
```
Comment (bm11100) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:10]:
```suggestion
Identifies use of the Defaults command to install a login or logoff hook in MacOS. An adversary may abuse this capability to establish persistence in an environment by 
inserting code to be executed at login or logout.
```

Comment (bm11100) [reply rules/macos/persistence_login_logout_hooks_defaults.toml:10]: will need to re lint

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Persistence via Login and/or Logout Hooks"
```
Comment (bm11100) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:16]:
```suggestion
name = "Persistence via Login or Logout Hook"
```
there can only be one Login hook at a time so `and` wasn't needed

**Review by bm11100 — APPROVED** [review]

File: rules/macos/persistence_login_logout_hooks_defaults.toml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+]
+risk_score = 47
+rule_id = "5d0265bf-dea9-41a9-92ad-48a8dcd05080"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence"]
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+ process.name : "defaults" and process.args : "write" and process.args : ("LoginHook", "LogoutHook")
```
Comment (brokensound77) [inline rules/macos/persistence_login_logout_hooks_defaults.toml:29]:
```suggestion
 process.name == "defaults" and process.args == "write" and process.args in ("LoginHook", "LogoutHook")
```

**Review by brokensound77 — APPROVED** [review]
One comment and LGTM

**Pull request merged**